packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +70,261 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-    fs::read_to_string(path).map_err(|e| Error::IO(path.into(), e))
+    fs::read_to_string(path).map_err(|err| Error::IO(path.into(), err))
 }
 
-fn eval_file<'m>(mv8: &'m MiniV8, path: &str) -> Result<mv8::Value<'m>> {
-    let code = read_file(path)?;
-    let v: mv8::Value<'_> = mv8
-        .eval(code)
-        .map_err(|e| Error::Eval(path.into(), e.into()))?;
-    Ok(v)
+fn eval_file<'s>(scope: &mut v8::HandleScope<'s>, path: &str) -> Result<v8::Local<'s, v8::Value>> {
+    let source_code = read_file(path)?;
+    let js_source_code = new_js_string(scope, &source_code)?;
+    let script = v8::Script::compile(scope, js_source_code, None)
+        .ok_or_else(|| Error::Eval(path.into(), "compile error".to_string()))?;
+
+    script
+        .run(scope)
+        .ok_or_else(|| Error::Eval(path.into(), "execution error".to_string()))
 }
 
-fn import_file<'m>(
-    mv8: &'m MiniV8,
+fn import_file<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    context: v8::Local<'s, v8::Context>,
     path: &str,
-    args: Vec<&mv8::Value<'m>>,
-) -> Result<mv8::Value<'m>> {
-    let v = eval_file(mv8, path)?;
-    let f = v
-        .as_function()
-        .ok_or_else(|| Error::FileImport(path.into(), format!("Failed to wrap file: {:?}", &v)))?;
-
-    let args = {
-        let mut a = Vec::new();
-        for arg in args {
-            a.push(arg.clone());
-        }
-        a
-    };
-    let args = mv8::Values::from_vec(args);
+    args: &[v8::Local<'s, v8::Value>],
+) -> Result<v8::Local<'s, v8::Value>> {
+    let v = eval_file(scope, path)?;
+    let f: v8::Local<'_, v8::Function> = v.try_into().map_err(|err| {
+        Error::FileImport(path.into(), format!("Failed to wrap file: {v:?}, {err}"))
+    })?;
+
+    let global_context = context.global(scope);
     let imported = f
-        .call(args)
-        .map_err(|e| Error::FileImport(path.into(), e.into()))?;
+        .call(scope, global_context.into(), args)
+        .ok_or_else(|| Error::FileImport(path.into(), "could not call function".to_string()))?;
 
     Ok(imported)
 }
 
-impl<'m> Embedded<'m> {
-    fn import(mv8: &'m MiniV8) -> Result<Self> {
-        let arrow = eval_file(mv8, "./src/worker/runner/javascript/apache-arrow-bundle.js")?;
-        let hash_stdlib = eval_file(mv8, "./src/worker/runner/javascript/hash_stdlib.js")?;
-        let hash_util = import_file(mv8, "./src/worker/runner/javascript/hash_util.js", vec![
-            &arrow,
-        ])?;
-        let batches_prototype = import_file(mv8, "./src/worker/runner/javascript/batch.js", vec![
-            &arrow, &hash_util,
-        ])?;
-
-        let ctx_import = import_file(mv8, "./src/worker/runner/javascript/context.js", vec![
-            &hash_util,
-        ])?;
-        let ctx_import = ctx_import.as_array().ok_or_else(|| {
+impl<'s> Embedded<'s> {
+    fn import(
+        scope: &mut v8::HandleScope<'s>,
+        context: v8::Local<'s, v8::Context>,
+    ) -> Result<Self> {
+        fn get_function_from_aray<'s>(
```

Comment:
Typo. Array*

Reply:
Why do we have this function and use it specifically here but not for the ones above where we extract functions from the array for packages (not runner.js)?

Reply:
This function is here to reduce code duplication, there was 7 times the same code https://github.com/hashintel/hash/pull/414#discussion_r833158808
We could add more helper functions, the `Option` everywhere make everything more verbose.

Reply:
Yes but I mean we do a similar thing where we get functions from an array on line 121 and we don't use this function. Right?

Reply:
Yes but the errors very different no?

Reply:
I mean I feel like a change in the function could address that, by adding the error context at the point the function is called rather than inside it. But I don't feel strongly